CMakeLists.txt

```diff
@@ -2,93 +2,173 @@ cmake_minimum_required (VERSION 3.5)
-#  Configure OpenMP
-find_package (OpenMP)
-set (CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${OpenMP_Fortran_FLAGS}")
-set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_Fortran_FLAGS}")
-option (USE_OMP "Use OpenOMP" OFF)
-if (${USE_OMP})
-    add_definitions (-DOMP)
-endif()
+#  Make sure the prefix path appears as a ccmake variable.
+set (CMAKE_PREFIX_PATH "" CACHE PATH "Search paths for dependency look up.")
 
+#-------------------------------------------------------------------------------
+#  Build Options
+#-------------------------------------------------------------------------------
 option (USE_ACC "Use OpenACC" OFF)
-if (${USE_ACC})
-   add_definitions (-DACC)
-endif()
-
-#  Configure MPI
-find_package (MPI REQUIRED)
-set (CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${MPI_Fortran_COMPILE_FLAGS}")
-set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${MPI_Fortran_LINK_FLAGS}")
-include_directories (${MPI_Fortran_INCLUDE_PATH})
+option (USE_PSPLINE "Use PSPLINE interpolation" OFF)
+option (USE_FIO "Use FIO interpolation support" OFF)
 
-#  Configure HDF5
-find_package (HDF5 REQUIRED COMPONENTS Fortran)
-include_directories (${HDF5_INCLUDE_DIRS})
+#-------------------------------------------------------------------------------
+#  Sanitizer options
+#-------------------------------------------------------------------------------
+add_library (sanitizer INTERFACE)
+
+macro (register_sanitizer_option name)
+    string (TOUPPER ${name} upper_name)
+    
+    option (SANITIZE_${upper_name} "Enable the ${name} sanitizer" OFF)
+
+    target_compile_options (sanitizer
+                            INTERFACE
+                            $<$<BOOL:${SANITIZE_${upper_name}}>:-fsanitize=${name}>
+    )
+    target_link_options (sanitizer
+                         INTERFACE
+                         $<$<BOOL:${SANITIZE_${upper_name}}>:-fsanitize=${name}>
+    )
+endmacro ()
+
+register_sanitizer_option (address)
+register_sanitizer_option (leak)
+register_sanitizer_option (memory)
+register_sanitizer_option (thread)
+register_sanitizer_option (undefined)
+register_sanitizer_option (float-divide-by-zero)
+
+#-------------------------------------------------------------------------------
+#  Build Locations
+#-------------------------------------------------------------------------------
+#  FIXME: Leave this as is for now but check if we want to use the cmake install
+#         functions better.
 
-#  Configure EZspline
-option (USE_PSPLINE "Use PSPLINE interpolation" OFF)
-if (${USE_PSPLINE})
-   #include_directories (${EZSpline_INCLUDE_PATH})
-   #set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${EZSpline_LINK_FLAGS}")
-   add_definitions (-DPSPLINE)
-endif ()
+#  Set directories to build all binary files.
+set (EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR}/build/bin)
+set (CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/build/modules)
+set (CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/build/lib)
```

Comment:
Do we need a second `build` directory inside `CMAKE_CURRENT_BINARY_DIR`?

I think [PROJECT_SOURCE_DIR](https://cmake.org/cmake/help/latest/variable/PROJECT_SOURCE_DIR.html) is a bit more precise.

Reply:
I think this is a problem from allowing in source builds. We should probably restrict that but currently, the CI isn't building KORC inside a build directory.

Reply:
Let's fix this antipattern in issue #14